__tests__/app.test.js

```diff
@@ -156,3 +158,103 @@ describe('/api/articles/:article_id', () => {
+                const {comments} = body;
+                expect(comments).toBeArray();
+                comments.forEach(comment => {
+                    expect(comment.article_id).toBe(1);
+                })
+            })
+    })
+    it('GET - status 200 - each comment should have the certain properties.', () => {
+        return request(app)
+            .get('/api/articles/1/comments')
+            .expect(200)
+            .then(({body}) => {
+                const {comments} = body;
+                const commentTemplate = {
+                    comment_id: expect.any(Number),
+                    votes: expect.any(Number),
+                    created_at: expect.any(String),
+                    author: expect.any(String),
+                    body: expect.any(String),
+                    article_id: expect.any(Number),
+                }
+                comments.forEach(comment => {
+                    expect(comment).toMatchObject(commentTemplate);
+                })
+            })
+    })
+    it('GET - status 200 - comments should be served with the most recent comments first.', () => {
+        return request(app)
+            .get('/api/articles/1/comments')
+            .expect(200)
+            .then(({body}) => {
+                const {comments} = body;
+                expect(comments).toBeSortedBy('created_at', {descending: true});
+            })
+    })
+    it('GET - status 200 - existing article_id without comments will return an empty array.', () => {
+        return connection
+            .query(`
+                SELECT
+                articles.article_id, COUNT(comments.*) AS comment_count
+                FROM articles
+                LEFT JOIN comments USING (article_id)
+                GROUP BY articles.article_id
+            `)
+            .then(({rows}) => {
+                return rows.filter(article => +article.comment_count === 0)
+            })
+            .then(articlesWithoutComments => {
+                return articlesWithoutComments.map(article => article.article_id)
+            })
+            .then(listOfArticlesWithoutComments => {
+                let index = 0;
+                let length = listOfArticlesWithoutComments.length
+
+                if (length === 0) {
+                    return;
+                } else if (length === 1) {
+                    index = index;
+                } else {
+                    index = Math.ceil(Math.random() * length)
+                }
+                return listOfArticlesWithoutComments[index];
+            })
+            .then((id) => {
+                if (id === undefined) return;
+                return request(app)
+                    .get(`/api/articles/${id}/comments`)
+                    .expect(200)
+                    .then(({body}) => {
+                        const {comments} = body;
+                        expect(comments).toBeArrayOfSize(0); 
+                    })
+            })
+    })
```

Comment:
Hey Kami, for this test you shouldn't need to go these lengths! 
Could you just refactor it so it has a bit more of a consistent look to your other tests? You should just be able to make a request to the app for an article with no comments (article 1 should work!) and then expect it to return an empty array. 🙂

Reply:
eh, article_id has 11 comments! in our database 2 perhaps should be fine... if beforeEach will help us reset everything that i need not to worry... Thanks! now going to erase that part :D

Reply:
@johnhutcheon  John... if i used article_id 1, it will fail the test as it has 11 comments by default setting in seed file
so i just instead of using 1, i used article_id 2.... but should works according to what i understood about what you wanted....? to hardcode the selected article_id which is existing but without comments, returning empty array